core/src/fields.rs

```diff
@@ -773,13 +773,18 @@ impl MongoStatement for MongoFields {
                 SqlDataType::SQL_VARCHAR
                 | SqlDataType::SQL_WVARCHAR
                 | SqlDataType::SQL_VARBINARY => match get_meta_data()?.char_octet_length {
-                    None => Bson::Int32(definitions::SQL_NO_TOTAL as i32),
-                    Some(char_octet_length) => Bson::Int32(char_octet_length as i32),
+                    None => Bson::Int32(
+                        i32::try_from(definitions::SQL_NO_TOTAL).expect("Could not convert to i32"),
```

Comment:
Something I notice about this one is that we use it in 3 places and cast it to an `i32` every time. Perhaps instead of casting, we should just define the const as an `i32` instead of a `Len`:
```
pub const SQL_NO_TOTAL: i32 = -4;
```

I suspect this will apply to several constants that we always cast. I'll call them out when I see them below.